.github/workflows/build-test.yml

```diff
@@ -67,34 +67,12 @@ jobs:
         with:
           path: ./*
           key: pre-build-${{ github.sha }}
-  docker-artifacts:
-    name: Build airnode-artifacts docker image
+  docker-build:
+    name: Build all the Docker containers and push them to Docker Hub
     runs-on: ubuntu-latest
+    # Don't run twice for a push within an internal PR
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
     needs: pre-build
-    steps:
-      - uses: actions/cache@v3
-        id: cache-pre-build
-        with:
-          path: ./*
-          key: pre-build-${{ github.sha }}
-      - name: Build airnode-artifacts
-        run: |
-          docker build --tag api3/airnode-artifacts-dev:${{ github.sha }} --file docker/Dockerfile .
-      - name: Login to DockerHub
-        uses: docker/login-action@v2
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Push airnode artifacts to Docker Hub
-        run: |
-          docker push api3/airnode-artifacts-dev:${{ github.sha }}
-  docker-build-general:
-    name: Build docker image
-    runs-on: ubuntu-latest
-    needs: docker-artifacts
-    strategy:
-      matrix:
-        package: [deployer, admin, client]
```

Comment:
I was a bit concerned about build times when I saw the matrix strategy removed, but it doesn't appear to be too affected:

This run: 16m 19s

Some before this:
- 13m 44s
- 15m 26s
- 15m 3s
- 14m 32s
- 16m 42s

Reply:
The matrix strategy was applied only to the builds for the final containers. Those builds are quite quick anyway so parallelizing them won't bring that much actually. The same goes for the new setup. And since there's no intermediate container (`airnode-artifacts`), I can't easily run these builds in a separate job. I'll figure out how to optimize it if necessary but I think it's fine for now.